src/theme-provider.js

```diff
@@ -71,7 +84,7 @@ ThemeProvider.contextTypes = {
 }
 
 ThemeProvider.propTypes = {
-  theme: PropTypes.object.isRequired,
+  theme: PropTypes.oneOfType([PropTypes.shape({}), PropTypes.func]).isRequired,
```

Comment:
I think `PropTypes.oneOfType([PropTypes.object, PropTypes.func]).isRequired` will be same, I just copied `theming` propTypes